### ISSUE TITLE
prevent creating foreign key by convention or navigation property

### ISSUE THREAD

**Issue by namighajiyev** [dotnet/efcore#27648]
My question is whether there is a way in Fluent API to prevent creation/discover of foreign relation by convention.
This is an example code below where I need to have such functionality. In the example below, I would like to have exactly same table created but having MyClass2 without any foreign key constraint to Company.

```
//base class
public abstract class MyBaseClass {
public int Id {get;set}
public int CompanyId Id {get;set}
public int Company Company {get;set}
}

//ok to have relations with Company
public  class MyClass1 :MyBaseClass {}

//ok to have CompanyId but don't want to have relations with Company
public  class MyClass2 :MyBaseClass {}
```

I am looking for something like this :
` modelBuilder.Entity<MyClass2>.PreventForeignKey(e=>e.CompanyId //or e.Company)`;

**Comment by namighajiyev** [comment]
@ajcvickers thanks for your answer. 

- I want **CompanyId** to be generated at database so removing it is not an option.
- Ignoring it like this `modelBuilder.Entity<MyClass2 >().Ignore(e => e.CompanyId);`  will remove it from database but I would like column to be there but without any relationship attached to it. 
- About acting as navigation that is not needed the property just comes from base class and not needed for any navigation purposes.

**Comment by ajcvickers** [comment]
@ajcvickers I'm not suggesting you remove/ignore the FK property (`CompanyId`), but rather the navigation (`Company`).

**Comment by namighajiyev** [comment]
@ajcvickers Ignoring navigation property  removed foreign key relation. Thank you a lot for the help.
 **modelBuilder.Entity<MyClass2>().Ignore(e => e.Company);**

**Comment by AlizerUncaged** [comment]
> [@ajcvickers](https://github.com/ajcvickers) Ignoring navigation property removed foreign key relation. Thank you a lot for the help. ` modelBuilder.Entity<MyClass2>().Ignore(e => e.Company);`

did this create the column?

**Comment by douglasaubre51** [comment]
i used ignore method in fluent it didnt work
i used notmapped that didnt work either
public class Channel
{
    [Key]
    public int ChannelId { get; set; }
    public string ChannelName { get; set; }
    public List<Account>? Accounts { get; set; }
}

public class Account
{
    [Key]
    public int AccountId { get; set; }
    [NotMapped]
    public Channel? channel { get; set; }
    public List<Message>? messages { get; set; }


    [Required]
    public string? Username { get; set; }
    [UniqueEmail]
    [DataType(DataType.EmailAddress)]
    public string? Email { get; set; }
    [Required]
    [DataType(DataType.Password)]
    public string? Password { get; set; }
    public string? ProfilePhotoURL { get; set; }


    public Roles roles { get; set; }
    public Departments departments { get; set; }
    public Semesters semesters { get; set; }
    public Status status { get; set; }
}
public class Message
{
    [Key]
    public int MessageId { get; set; }
    public string TextMessage { get; set; }
    public DateTime UploadDateTime { get; set; }
    [NotMapped]
    public Account? account { get; set; }
}

**Comment by uberubert** [comment]
@douglasaubre51 I had the same problem in my code-first setup, on a many-to-many relation. Not sure if my case matches yours perfectly, but adding ".Ignore()" did not remove the foreign key. I also tried making them nullable, adding notmapped, played with the modelbuilder in various ways... Nothing helped.

I solved it by doing 3 things:
- Changed the id property name to something that avoids the ef core naming conventions
- Removed the navigational property from the optional entity model
- Left the ".Ignore(relationalEntity => relationalEntity.OptionalEntity)"

Not sure if all 3 are necessary, and I have to fetch the optional entities in a more manual way now (which I expected to do anyway). But the foreign key is gone, and I didn't need to add ".Ignore()".
